### PR TITLE
[Feature] Restore Purchases

### PR DESCRIPTION
**Pull request by cb-amutha**
Adds Restore Purchases

### CODE REVIEW

File: android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt [hunk 1]
```diff
@@ -141,6 +145,28 @@ class ChargebeeFlutterSdkPlugin : FlutterPlugin, MethodCallHandler, ActivityAwar
             })
     }
 
+
+    private fun restorePurchases(resultCallback: Result, queryParams: Map<String, Boolean>?) {
+        val includeInActivePurchases = queryParams?.get("includeInActivePurchase") as Boolean
```
Comment (cb-haripriyan) [inline android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt:150]: Let's rename it to `includeInactivePurchases` in all places we use in flutter.

File: android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt [hunk 2]
```diff
@@ -141,6 +145,28 @@ class ChargebeeFlutterSdkPlugin : FlutterPlugin, MethodCallHandler, ActivityAwar
...
+            activity,
+            includeInActivePurchases,
+            object : CBCallback.RestorePurchaseCallback {
+                override fun onSuccess(result: List<CBRestoreSubscription>) {
+                    restoreSubscription.clear()
+                    result.forEach { subscription ->
+                        val jsonMapString = Gson().toJson(subscription.toMap())
+                        restoreSubscription.add(jsonMapString)
+                    }
+                    resultCallback.success(restoreSubscription)
```
Comment (cb-haripriyan) [inline android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt:161]: Do we need to keep `restoreSubscription` at instance level? Can we refactor to have it at method level?

File: README.md [hunk 3]
```diff
@@ -109,13 +109,49 @@ try {
...
+| 2018         | This error occurs when the data contained within the receipt is not valid or cannot be parsed.                          |
+| 2019         | This error occurs when there are no products available to restore.                             |
+| 2020         | This error occurs when there is an error with the Chargebee service during the restore process.
+| -3            | The request has reached the maximum timeout before Google Play responds.       |
+| -2 | The requested feature is not supported by the Play Store on the current device.                                             |
+| -1         | The app is not connected to the Play Store service via the Google Play Billing Library.                             |
+| -4         | Unknown error occurred.
+| 2        | The service is currently unavailable.         |
+| 5  | Error resulting from incorrect usage of the API.                                                          |
+| 6         | Fatal error during the API action.                             |
```
Comment (cb-haripriyan) [inline README.md:146]: Can we standardize the error codes across android and ios?

File: lib/src/models/product.dart [hunk 4]
```diff
@@ -9,7 +9,7 @@ class Product {
   /// Currency code for the price
   late String currencyCode;
   /// Local currency price for the product in double
-  late double price;
+  late num price;
```
Comment (cb-haripriyan) [inline lib/src/models/product.dart:12]: would this need any changes at the customer's logic?

File: lib/src/models/restore_subscription.dart [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import 'package:flutter/cupertino.dart';
+
+class CBRestoreSubscription{
+  /// subscriptionId
+  late String subscriptionId;
+  /// planId
+  late String planId;
+  /// subscription status on store
+  late String storeStatus;
```
Comment (cb-haripriyan) [inline lib/src/models/restore_subscription.dart:9]: Should this be an `enum`?

**Review by cb-haripriyan — CHANGES_REQUESTED** [review]

File: lib/src/models/restore_subscription.dart (hunk 5 above)
Comment (cb-amutha) [reply lib/src/models/restore_subscription.dart]: Done!

File: lib/src/models/product.dart (hunk 4 above)
Comment (cb-amutha) [reply lib/src/models/product.dart:12]: No impacts on customer logic.

File: android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt (hunk 2 above)
Comment (cb-amutha) [reply android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt:162]: Done

File: android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt [hunk 6]
```diff
@@ -371,3 +406,46 @@ fun CBProduct.periodUnit(): String {
...
+                BillingClient.BillingResponseCode.USER_CANCELED -> USER_CANCELED
+                BillingClient.BillingResponseCode.BILLING_UNAVAILABLE -> BILLING_UNAVAILABLE
+                BillingClient.BillingResponseCode.ITEM_UNAVAILABLE -> ITEM_UNAVAILABLE
+                BillingClient.BillingResponseCode.ITEM_NOT_OWNED -> ITEM_NOT_OWNED
+                BillingClient.BillingResponseCode.ITEM_ALREADY_OWNED -> ITEM_ALREADY_OWNED
+                else -> UNKNOWN
+            }
+        }
+    }
+}
```
Comment (cb-haripriyan) [inline android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt:450]: Can the error codes be moved to a new file accordingly?

File: ios/Classes/SwiftChargebeeFlutterSdkPlugin.swift [hunk 7]
```diff
@@ -258,7 +281,16 @@ extension SKProduct {
...
+extension InAppSubscription{
+    func toMap() -> [String: Any?] {
+        let map: [String: Any?] = [
+            "subscriptionId": subscriptionID,
+            "planId": planID,
+            "storeStatus": storeStatus.rawValue
+        ]
+        return map
+    }
+}
```
[the remark this answers is not in the document]
Comment (cb-amutha) [reply ios/Classes/SwiftChargebeeFlutterSdkPlugin.swift]: Moved to new file. thanks

File: android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt (hunk 6 above)
Comment (cb-amutha) [reply android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt]: yes, Done!

**Review by cb-haripriyan — APPROVED** [review]

**Pull request merged**